autogluon/task/tabular_prediction/tabular_prediction.py

```diff
@@ -56,6 +56,7 @@ def load(output_directory, verbosity=2):
     @unpack(set_presets)
     def fit(train_data,
             label,
+            unlabeled_data=None,
```

Comment:
Can we include a sanity check that `unlabeled_data` is actually of the appropriate format? (i.e. contains the same columns as `train_data`), and print an informative error msg if not?

Reply:
@Innixma Added some minor docs for this, will need to do more research to make it a bit more descriptive. Added a TODO.
@jwmueller I don't have a problem doing this column check, but are we mandating that the unlabeled data must be 1:1 column-wise to the training data? Was my thought that it would be more flexible than that, but I could be wrong.

Reply:
Upon further discussion with @jwmueller, we aren't currently dealing with schema mismatches. I'll add the unlabeled_data/train_data check.